althea_kernel_interface/src/exit_server_tunnel.rs

```diff
@@ -75,14 +74,30 @@ impl dyn KernelInterface {
         // only get the flows list once
         let flows = self.get_flows(if_name)?;
         for c in clients.iter() {
+            // Add ipv4 flows
+            let ipv4;
             match c.internal_ip {
                 IpAddr::V4(addr) => {
+                    ipv4 = addr;
                     if !self.has_flow_bulk(addr, &flows) {
-                        self.create_flow_by_ip(if_name, addr)?
+                        self.create_flow_by_ip(if_name, addr)?;
                     }
                 }
                 _ => panic!("Could not derive ipv4 addr for client! Corrupt DB!"),
             }
+
+            // Add ipv6 flows
+            let ipv6_list = c.internet_ipv6_list.clone();
+            let ipv6_list: Vec<&str> = ipv6_list.split(',').collect();
+
+            for ip in ipv6_list {
+                // Verfiy its a valid subnet
+                if let Ok(ip_net) = ip.parse::<IpNetwork>() {
```

Comment:
can we do the validation further up? maybe with a utility function to convert the response from the database and log errors? In this case if there are any problems with the data in the db we just silently ignore it. Ideally we can do verification and then biz logic, not mix them.

Reply:
Fixed